### PR TITLE
ESC key closes upload modal.

### PR DESCRIPTION
**Pull request by prastut**
Note:
* Created a custom directive for keypress events for future usage. In future if we need other events like keypress, keydown we can add it to the directive.
* Added a function listening to the above directive in `uploadPictureModal.js` directive. 
* Added the custome directive to the HTML template `upload-picture-modal.js`. 

For issue #183

### CODE REVIEW

**Comment by llopv** [comment]
I'm wondering if you could hack it a little more to have a directive like `close-on-esc="uploadPictureSharedState"`. If you don't want to do it right now, I would be happy to merge what you have anyway.

**Comment by prastut** [comment]
@llopv Willing to hack on it. 

Updated the PR with a more generic directive. Some comments about the same:

* I like your approach about having something like `close-on-esc ="uploadPictureSharedState"` . That way it becomes self-explanatory. 
* But since my directive is a generic key event listener, I don't want to make it state specific. It should ideally only deal with keypress events and send the keypress event to the controller which is managing that particular template. The controller can do whatever it feels like with the event.

Would want to know your opinion on this.

File: src/templates/upload-picture-modal.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="modal modal-overlay" ui-if="uploadPictureSharedState" ui-state="uploadPictureSharedState" key-press>
+<div class="modal modal-overlay" ui-if="uploadPictureSharedState" ui-state="uploadPictureSharedState" key-press="keyup">
```
Comment (prastut) [inline src/templates/upload-picture-modal.html]: Now I can send events based on the key event I want. The directive could take values like:

`key-press="keyup"`
`key-press="keydown"`
`key-press="keypress"`

Comment (prastut) [inline src/templates/upload-picture-modal.html]: Now I can send events based on the key event I want. The directive could take values like:

key-press="keyup"
key-press="keydown"
key-press="keypress"

**Pull request merged**